source/slang/slang-shader-cache-index.h

```diff
@@ -0,0 +1,65 @@
+#pragma once
+#include "../../slang.h"
+
+#include "../core/slang-string.h"
+#include "../core/slang-linked-list.h"
+#include "../core/slang-dictionary.h"
+
+namespace Slang
+{
+    class ShaderCacheIndex
+    {
+    public:
+        ShaderCacheIndex(SlangInt size)
+            : entryLimit(size)
+        {}
+
+        // Load a previous cache index saved to disk. If not found, create a new cache index
+        // and save it to disk as filename.
+        SlangResult loadCacheIndex(String filename);
+
+        // Fetch the cache entry corresponding to the provided key. If found, store the
+        // mapped AST hash for that entry in outContents and return SLANG_OK. Else, set
+        // outContents to nullptr and return SLANG_FAIL.
+        SlangResult fetchEntry(String key, String* outContents);
+
+        // Add an entry to the cache with the provided key and contents hashes. If
+        // adding an entry causes the cache to exceed size limitations, this will also
+        // delete the least recently used entry.
+        void addEntry(String key, String contents);
+
+        // Update the contents hash for the specified entry in the cache.
+        void updateEntry(String key, String newContents);
+
+    private:
+        // Update the entry for key in the cache index on disk. This should be called any
+        // time keyToContents changes.
+        SlangResult updateCacheIndex(String key);
+
+        // Move the entry corresponding to key to the front of entries. This should only
+        // be called by fetchEntry upon successfully finding an existing entry.
+        void moveEntryToFront(String key);
```

Comment:
Shouldn't need this. You do the lookup in `findEntry` and then immediately move the `LInkedNode` to front there. If you prefer to have this function, the signature should be `moveEntryToFront(LinkedNode<ShaderCacheIndexEntry>*)` to prevent doing extra lookups.